[add] the theme configuration section (#18)
dd3tech/dd360-components-docs: docs/get-started/getting-started.mdx

<WindowEditor language="javascript" header={{ show: true, title: 'App.js' }} codeString="import 'dd360-ds/dd360.css'" />

<br />

## <span name="floating-nav">Setup</span>
For our library to work properly, you will need to wrap your application inside our Provider.

Go to <span className="px-1 py-0.5 bg-pink-100 rounded-md">pages/_app.js</span> or <span className="px-1 py-0.5 bg-pink-100 rounded-md">pages/_app.tsx</span> (create it if it doesn't exist) and add this:

### Next.js

> <WindowEditor language="typescript" codeString={`import { ThemeProvider } from 'dd360-ds/theme'

Reviewer: ```suggestion
<WindowEditor language="tsx" codeString={`import { ThemeProvider } from 'dd360-ds/theme'
```

import 'dd360-ds/dd360.css'

export default function App({ Component, pageProps }: AppProps) {
    return (
        <ThemeProvider>
            <Component {...pageProps} />
        </ThemeProvider>
    )
}`} />

### React.js

Go to the root of your application and do this:

> <WindowEditor language="typescript" codeString={`import { ThemeProvider } from 'dd360-ds/theme'

Reviewer: ```suggestion
<WindowEditor language="tsx" codeString={`import { ThemeProvider } from 'dd360-ds/theme'
```

import 'dd360-ds/dd360.css'

ReactDOM.createRoot(document.getElementById("root")).render(
  <StrictMode>

>     <ThemeProvider theme={theme}>

Reviewer: ```suggestion
    <ThemeProvider>
```

      <App />
    </ThemeProvider>
  </StrictMode>
);`} />

## <span name="floating-nav">Customize theme (optional) 🎨👨‍🎨</span>

ThemeProvider provides a simple way to customize the default themes, you can change the colors and fonts.

To extend or override a token in the default theme, import the <span className="px-1 py-0.5 bg-pink-100 rounded-md">createTheme</span> function and add the keys you want to override. You can also add new values to the theme.

For example, if you want to update your theme colors to include your brand colors, here's what to do:

> <WindowEditor language="typescript" codeString={`import { createTheme,  ThemeProvider, Text  } from "dd360-ds/theme" 

Reviewer: ```suggestion
<WindowEditor language="tsx" codeString={`import { createTheme,  ThemeProvider, Text  } from "dd360-ds/theme" 
```

<WindowEditor language="javascript" header={{ show: true, title: 'App.js' }} codeString="import 'dd360-ds/dd360.css'" />

<br />

> ## <span name="floating-nav">Provider</span>

Reviewer: ```suggestion
## <span name="floating-nav">Setup</span>
```

> For DD360-ds to work correctly, you need to set up the ThemeProvider at the root of your application.

Reviewer: ```suggestion
For our library to work properly, you will need to wrap your application inside our Provider.
```

> Go to the root of your application and do this:

Reviewer: I suggest we put an example of how it would look like in Nextjs and React 

```suggestion
Go to the root of your application and do it.

For Nextjs projects, you can do it as':
```

> <WindowEditor language="bash" codeString={`import * as React from 'react';

Reviewer: ```suggestion
<WindowEditor language="typescript" codeString={`import * as React from 'react';
```

// 1. import 'ThemeProvider'
import { ThemeProvider } from 'dd360-ds/theme'
import 'dd360-ds/dd360.css'

function App({ Component, pageProps }: AppProps) {
    // 2. Use at the root of your app
    return (
        <ThemeProvider>
            <Component />
        </ThemeProvider>
    )
}`} />

## <span name="floating-nav">Customize theme (optional) 🎨👨‍🎨</span>

ThemeProvider provides a simple way to customize the default themes, you can change the colors and fonts.

To extend or override a token in the default theme, import the <span className="px-1 py-0.5 bg-pink-100 rounded-md">createTheme</span> function and add the keys you want to override. You can also add new values to the theme.

For example, if you want to update your theme colors to include your brand colors, here's what to do:

> <WindowEditor language="bash" codeString={`import { createTheme,  ThemeProvider, Text  } from "dd360-ds/theme" 

Reviewer: ```suggestion
<WindowEditor language="javascript" codeString={`import { createTheme,  ThemeProvider, Text  } from "dd360-ds/theme" 
```

import { Text } from "dd360-ds

const theme = createTheme({
  // // brand colors
  palette: {
    primary: {
      contrastText: 'white',
      main: '#1c4ed9'
    },
    secondary: {
      main: 'white',
      contrastText: ''
    },
    success: {
      main: '#22c55e'
    },
    warning: {
      main: '#eab308'
    },
    info: {
      main: '#6b7280'
    },
    error: {
      main: '#ef4444'
    },
    background: '#fff',
    textColor: '#000'
  },
  typography: {
    fontFamily: 'Rubik Pixels',
    h1: { fontSize: 36 },
    h2: { fontSize: 32 },
    h3: { fontSize: 28 },
    h4: { fontSize: 24 },
    h5: { fontSize: 20 },
    h6: { fontSize: 18 },
    base: {
      fontSize: 16
    }
  }

  // 3. Pass the new theme to 'ThemeProvider'
  <ThemeProvider theme={theme}>
    <App />
  </ThemeProvider>

  // 4. Now you can use these colors in your components
  function MyComponent() {

>     return <Text className="text-primary">NextUI colors</Text>

Reviewer: ```suggestion
    return <Button>Button primary</Button>
```

<br />
<br />

## 💅 Customize theme

> If you want to use a provider to configure the library's theme, you can do so using the ThemeProvider provided by dd360-ds. This will allow you to set your own settings for the theme.

Reviewer: I think it should not really be optional to use the provider, otherwise it won't even load our default palette.

```suggestion
To use the library it's important that you wrap the application on our `Provider`. You can will customize with your own palette of colours and pass to our provider without complexity
```

Author: It is not really that important to use the provider, the application does not break because it has a default theme in the library

> To use the ThemeProvider, you must import it as follows:

Reviewer: ```suggestion
You can import the provider, like this:
```

> ```tsx

Reviewer: you left an empty opening declaration

```

<WindowEditor language="javascript" header={{ show: true, title: 'App.js' }} codeString="import { createTheme, ThemeProvider } from 'dd360-ds/theme'" />

> You can then create your own theme configuration using the createTheme function, which accepts an object with different properties to customize the theme. For example, you can define the primary and secondary colors as shown below:

Reviewer: ```suggestion
You can set your own theme configuration using the createTheme function, which accepts an object with different properties to customize the theme. For example, you can define the primary and secondary colors as shown below:
```

<WindowEditor
    language="javascript"
    header={{ show: true, title: 'App.js' }}
    codeString={`const theme = createTheme({
  palette: {
    primary: {
      main: "purple"
    },
    secondary: {
      main: "#FFC107"
    }
  }
})`}
/>

> After creating your theme configuration, you can use the ThemeProvider to wrap your React app. This can be done as follows:

Reviewer: ```suggestion
After set your theme configuration, you need wrap your app inside of the provider. This can be done as follows:
```

<WindowEditor
    language="javascript"
    header={{ show: true, title: 'App.js' }}
    codeString={`ReactDOM.createRoot(document.getElementById("root")).render(
    <StrictMode>
        <ThemeProvider theme={theme}>
        <App />
        </ThemeProvider>
    </StrictMode>
)`}
/>

> Within the ThemeProvider, you can use the components provided by dd360-ds and the corresponding styles will be applied based on your theme settings.

Reviewer: ```suggestion
Now with this, you will be able to customize all the colors of your palette and use our components with your own color settings. This brings some advantages such as being able to switch from one theme to another without complexity, like dark and light theme style.
```
